src/include/ipc/topology.h

```diff
@@ -28,6 +28,7 @@ enum sof_comp_type {
 	SOF_COMP_NONE = 0,
 	SOF_COMP_HOST,
 	SOF_COMP_DAI,
+	SOF_COMP_CHAIN_DMA,
```

Comment:
can we do this @ranj063 @jsarha ? This would break backwards-compatibility with the kernel. This should be added line 52

I also thought we'd stop adding new component types and use UUIDs now?

Reply:
Yes - we should use UUIDs now. IIUC Windows should also use an UUID for this ? @marcinszkudlinski ?

Reply:
My understanding of the topology system is still quite limited, but while browsing the code I could not immediately tell why this would not work with the current driver side implementation. The component IDs are not present in IPC4 messages.

I also tried to check this with brute force method by testing it, but the build failed with this error:

/home/oku/work/sof-zephyr-new/sof/src/ipc/ipc4/helper.c:593:10: fatal error: sof/mem_win_debug.h: No such file or directory
  593 | #include "sof/mem_win_debug.h"
      |          ^~~~~~~~~~~~~~~~~~~~~

Any idea what I am missing?

Reply:
Managed to build the branch by removeing the win debug code, but the FW still does not work. e.g. its not only HDMI audio that is broken, but nothing really works. I am probably missing something.

Reply:
SOF_COMP_CHAIN_DMA removed.
Resolved in:
![image](https://user-images.githubusercontent.com/93590538/203564943-42950cd5-4ff3-4a68-b200-ce227fed6c3e.png)